components/trusted-app-mgt/org.wso2.carbon.identity.trusted.app.mgt/src/main/java/org/wso2/carbon/identity/trusted/app/mgt/services/TrustedAppMgtServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.IOS_CREDENTIAL_PERMISSION;
+
+/**
+ * Trusted App Management Service Implementation.
+ */
+public class TrustedAppMgtServiceImpl implements TrustedAppMgtService {
+
+    private static final Log LOG = LogFactory.getLog(TrustedAppMgtServiceImpl.class);
+
+    @Override
+    public List<TrustedAndroidApp> getTrustedAndroidApps() throws TrustedAppMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Retrieving trusted apps for Android platform.");
+        }
+        List<TrustedAndroidApp> trustedAndroidApps = new ArrayList<>();
+        try {
+            List<TrustedApp> trustedAppSet =
+                    TrustedAppMgtDataHolder.getInstance().getApplicationManagementService()
+                            .getTrustedApps(PlatformType.ANDROID);
+            for (TrustedApp trustedApp : trustedAppSet) {
+                TrustedAndroidApp trustedAndroidApp = new TrustedAndroidApp();
+                trustedAndroidApp.setPackageName(trustedApp.getAppIdentifier());
+                try {
+                    trustedAndroidApp.setThumbprints(resolveThumbprints(trustedApp));
+                    trustedAndroidApp.setPermissions(resolveAppPermissions(trustedApp));
+                } catch (TrustedAppMgtException e) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(e.getMessage() + "Hence removing the app from the trusted app list.");
+                    }
+                    continue;
+                }
+                trustedAndroidApps.add(trustedAndroidApp);
+            }
+        } catch (IdentityApplicationManagementException e) {
+            throw new TrustedAppMgtException("Error occurred while retrieving trusted apps for " +
+                    "Android platform.", e);
+        }
+        return trustedAndroidApps;
+    }
+
+    @Override
+    public List<TrustedIosApp> getTrustedIosApps() throws TrustedAppMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Retrieving trusted apps for iOS platform.");
+        }
+        List<TrustedIosApp> trustedIosApps = new ArrayList<>();
+        try {
+            List<TrustedApp> trustedAppSet =
+                    TrustedAppMgtDataHolder.getInstance().getApplicationManagementService().
+                            getTrustedApps(PlatformType.IOS);
+            for (TrustedApp trustedApp : trustedAppSet) {
+                TrustedIosApp trustedIosApp = new TrustedIosApp();
+                trustedIosApp.setAppId(trustedApp.getAppIdentifier());
+                try {
+                    trustedIosApp.setPermissions(resolveAppPermissions(trustedApp));
+                } catch (TrustedAppMgtException e) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(e.getMessage() + "Hence removing the app from the trusted app list.");
```

Comment:
log the exception as well
```suggestion
                        LOG.debug(e.getMessage() + "Hence removing the app from the trusted app list.", e);
```

Reply:
better to make this a error log rather than a debug log

Reply:
This try catch block is removed and handled differently